.github/workflows/tuttest.yml

```diff
@@ -0,0 +1,102 @@
+name: tuttest
+
+on:
+  pull_request:
+    paths:
+    - 'binutils/*'
+    - '.github/workflows/tuttest.yml'
+    - 'README.md'
+  push:
+    paths:
+    - 'binutils/*'
+    - '.github/workflows/tuttest.yml'
+    - 'README.md'
+  workflow_dispatch:
+
+env:
+  ARTIFACT_NAME: tuttest-script
+  SCRIPT: tuttest-script.sh
+
+jobs:
+  grab-readme-commands:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - name: Install tuttest
+        run: |
+          python3 -m pip install setuptools
+          python3 -m pip install git+https://github.com/antmicro/tuttest@5dbe5845c9ef46f1c0315129449743db010ff966
+          # On GitHub Actions, pip installs "binaries" to this directory
+          echo "/home/runner/.local/bin" >> $GITHUB_PATH
+
+      - name: Test tuttest
+        run: tuttest --help
+
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Grab install-prerequisites commands with tuttest
+        run: |
+          echo "# README.md/install-prerequisites" >$SCRIPT
+          echo >>$SCRIPT
+          # SED changes URL in case the CI is running on a fork
+          tuttest README.md install-prerequisites | sed -e "s#git clone .*.git#git clone https://github.com/$GITHUB_REPOSITORY.git#" >>$SCRIPT
+          echo >>$SCRIPT
+          echo "# Checkout commands" >>$SCRIPT
+          echo >>$SCRIPT
+
+      - name: Add checkout commands for a Pull Request CI
+        if: ${{ github.event_name == 'pull_request' }}
+        run: |
+          echo "git fetch origin $GITHUB_REF:ci-pr-branch" >>$SCRIPT
+          echo "git checkout ci-pr-branch" >>$SCRIPT
+
+      - name: Add checkout commands for a non-Pull Request CI
+        if: ${{ github.event_name != 'pull_request' }}
+        run: |
+          echo "git checkout $GITHUB_SHA" >>$SCRIPT
+
+      - name: Grab prepare-and-build commands with tuttest
+        run: |
+          echo >>$SCRIPT
+          echo "# README.md/prepare-and-build" >>$SCRIPT
+          echo >>$SCRIPT
+          tuttest README.md prepare-and-build >>$SCRIPT
+
+      - name: Print ${{ env.SCRIPT }}
+        run: cat $SCRIPT
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: ${{ env.ARTIFACT_NAME }}
```

Comment:
Since there is a single artifact, setting the name is not necessary.

Reply:
It's good to know it isn't needed but I've left it unchanged because without a name all artifacts are downloaded to separate directories, even if there's a single one. I'd need to adapt to it or specify path during downloading so I think the current solution is better.